### ISSUE TITLE
Can edit be authorized based on role ability changed runtime 

### ISSUE THREAD

**Issue by swijal** [CanCanCommunity/cancancan#232]
I have overridden current_ability method in application controller as below to pass @role (Set of roles) instead of current_user.

def current_ability
  @current_ability ||= Ability.new @role
end

Initially I am initializing @role in Application controller based on some predefined conditions.
if var1 == "mine"
  @role.add("abc")
end

Now I need to add one additional role say team_role in ability class as below but not to initialize in application controller but in controller1 as shown below:

Ability class:
if @role.include?("abc")
can :manage, model1
end 
if @role.include?("team_role")
can :edit, model1
end 

But the role team_role will be defined runtime in controller1 as shown below after calling some function in show (I am not sure if that is possible):

controller1 class:

class Controller1 < ApplicationController
load_and_authorize_resource :model1, :only => :edit

def show
 loads data from DynamoDB using model1 and later on passed to show.html to just show the values in the fields.
  @param1 = funct1.search search_params
  if @param1.size == "L"
    @role.add("team_role")
  end
end

def edit
show
 calls show and later on passed to edit.haml to show values in editable fields.
end

end

Will that authorize edit  based on the condition defined runtime in controller1 if size is "L"?

In short, I want to be able to add the @role runtime in show method so that when edit link will be hit then it will authorize and based on that check it should authorize
As of now it is not authorizing.

Thanks

**Comment by swijal** [comment]
Never mind...I can do it by checking abilities for block but the problem was with the class that I need to provide as we don't have models linked with controller so which  class to pass. We were using java services and I passed Java class and it worked
